devicetypes/smartthings/zigbee-motion-sensor-light.src/led-cpx-light.groovy

```diff
@@ -46,49 +50,44 @@ metadata {
 				attributeState "level", action: "switch level.setLevel"
 			}
 		}
-		
+
 		standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
 			state "default", label: "", action: "refresh.refresh", icon: "st.secondary.refresh"
 		}
 		
 		controlTile("colorTempSliderControl", "device.colorTemperature", "slider", width: 4, height: 2, inactiveLabel: false, range: "(2700..6500)") {
 			state "colorTemperature", action: "color temperature.setColorTemperature"
 		}
-		
-		valueTile("colorName", "device.colorName", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
-			state "colorName", label: '${currentValue}'
-		}
-		
+        
 		main(["switch"])
-		details(["switch", "refresh", "switchLevel", "colorTempSliderControl", "colorName"]) //
+		details(["switch", "switchLevel", "colorTempSliderControl", "refresh"])
 	}
 }
 
-private getMOTION_CLUSTER_VALUE() { 0x0406 }
-private getMOTION_STATUS_VALUE() { 0x0000 }
-private getON_OFF_CLUSTER_VALUE() { 0x0006 }
-
-// Parse incoming device messages to generate events
 def parse(String description) {
 	def event = zigbee.getEvent(description)
-	def zigbeeMap = zigbee.parseDescriptionAsMap(description)
+	
 	if (event) {
 		if (!(event.name == "level" && event.value == 0)) {
 			sendEvent(event)
 		}
-	} else if (zigbeeMap.cluster == MOTION_CLUSTER_VALUE && zigbeeMap.attrId == MOTION_STATUS_VALUE) {
-		def childDevice = getChildDevices()?.find {
-			it.device.deviceNetworkId == "${device.deviceNetworkId}:1"
-		}
-		zigbeeMap.value = zigbeeMap.value.endsWith("01") ? "active" : "inactive"
-		zigbeeMap.name = "motion"
-		childDevice.sendEvent(zigbeeMap)
 	} else {
 		def cluster = zigbee.parse(description)
-		if (cluster && cluster.clusterId == ON_OFF_CLUSTER_VALUE && cluster.command == 0x07) {
+		if (cluster && cluster.clusterId == 0x0006 && cluster.command == 0x07) {
 			if (cluster.data[0] == 0x00) {
 				sendEvent(name: "checkInterval", value: 60 * 12, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
 			}
+		} else {
+			def zigbeeMap = zigbee.parseDescriptionAsMap(description)
+			
+			if (zigbeeMap.cluster ==  "0406" && zigbeeMap.attrId == "0000") {
```

Comment:
you can use, respectively, `zigbeeMap.clusterInt` and `zigbeeMap.attrInt` and use previously defined constants, instead of coming back to the mediocre solution

Reply:
Great! It works well. Thank you very much.